### PR TITLE
Concat and minify css and js files

### PR DESCRIPTION
**Pull request by iraquitan**
Add gulp task to handle css and js concat and minify as stated at #22 .. if you have any doubts or improvements, just ask.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        './source/vendor/lightgallery/js/lg-zoom.min.js',
+        './source/vendor/lightgallery/js/lg-hash.min.js',
+        './source/vendor/lightgallery/js/lg-share.min.js',
+        './source/vendor/lightgallery/js/lg-video.min.js',
+        './source/vendor/justified-gallery/jquery.justifiedGallery.min.js',
+        './source/js/main.js',
+        './source/js/contact.js',
+        './source/js/form.js'
+    ])
+        .pipe(concat('vendor-scripts.min.js'))
```
Comment (butlerx) [inline gulpfile.js]: not a fan of the name as its not just vendor scripts id prefer just calling it main

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+gulp.task('vendor-styles', function () {
+    gulp.src([
+        './source/vendor/font-awesome/css/font-awesome.min.css',
+        './source/vendor/open-sans/styles.css',
+        './source/vendor/source-code-pro/styles.css',
+        './source/vendor/lightgallery/css/lightgallery.min.css',
+        './source/vendor/justified-gallery/justifiedGallery.min.css'
+        ])
+        .pipe(concat('vendor-styles.min.css'))
+        .pipe(cleanCSS())
```
Comment (butlerx) [inline gulpfile.js]: could you use these options `.cleanCSS({compatibility: 'ie9', processImport: false, keepSpecialComments: 0})`

**Review by butlerx — CHANGES_REQUESTED** [review]

File: gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+var gulp = require('gulp');
+var concat = require("gulp-concat");
+var uglify = require("gulp-uglify");
+var cleanCSS = require('gulp-clean-css');
+
+gulp.task('minify-all', ['vendor-styles', 'vendor-fonts', 'vendor-images', 'scripts']);
```
Comment (butlerx) [inline gulpfile.js]: could you change this to default

File: package.json [hunk 4]
```diff
@@ -3,6 +3,10 @@
   "version": "0.2.1",
   "private": true,
   "dependencies": {
```
Comment (butlerx) [inline package.json:8]: add a postinstall to run gulp

**Review by butlerx — CHANGES_REQUESTED** [review]
all so i know it wasnt part of the original spec but could you add gulp-bower to install the power components and add `.source/vendor` to the .gitignore

File: package.json [hunk 5]
```diff
@@ -3,6 +3,10 @@
...
-    "desandro-classie": "^1.0.1"
+    "desandro-classie": "^1.0.1",
```
Comment (butlerx) [inline package.json]: also this is no longer used and can be removed i believe

**Comment by iraquitan** [comment]
Added the requested changes.

File: layout/common/scripts.ejs [hunk 6]
```diff
@@ -1,7 +1,4 @@
 <%- partial('comment/scripts', { page: page }) %>
-<%- partial('plugin/scripts') %>
 
 <!-- Custom Scripts -->
-<%- js('js/main') %>
-<%- js('js/contact') %>
-<%- js('js/form') %>
+<%- js('vendor/dest/js/vendor-scripts.min') %>
```
Comment (butlerx) [inline layout/common/scripts.ejs]: this needs to be main.js

**Review by butlerx — CHANGES_REQUESTED** [review]

**Comment by iraquitan** [comment]
Fixed the js name and import

**Review by butlerx — APPROVED** [review]

**Comment by butlerx** [comment]
thanks for the help

**Pull request merged**